pkg/pillar/hypervisor/xen.go

```diff
@@ -849,28 +849,21 @@ func isDomainRunning(domainID int) bool {
 	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
 	if err != nil {
 		//domain is not present
-		log.Errorln("verifyDomainState: xl list failed ", err)
-		log.Errorln("verifyDomainState: xl list output ", string(stdoutStderr))
-		return false
+		log.Errorln("checkDomainHealthy: xl list failed ", err)
+		log.Errorln("checkDomainHealthy: xl list output ", string(stdoutStderr))
+		return err
 	}
 	//Removing all extra space between column result and split the result as array.
 	xlDomainResult := regexp.MustCompile(`\s+`).ReplaceAllString(strings.Split(string(stdoutStderr), "\n")[1], " ")
 	//Domain's status is 5th column in xl list <domain> result
 	domainState := strings.Split(xlDomainResult, " ")[4]
-	//Sometimes the domain can be in multiple states (logically possible) at the same time.
-	// In such cases we can consider the last state. For example when a domain is shutting down its state will be
-	//"--psc-", we can consider "c" to represent that domain is in middle of shutting down.
-	var lastState string
+	log.Debugf("checkDomainHealthy: domain: %d domainState: %s.", domainID, domainState)
+	//if domainState has any of the these bit set ['s', 'c', 'd'] then we can consider that the domain is not running
 	for _, state := range domainState {
-		if state != '-' {
-			lastState = string(state)
+		if state == 's' || state == 'c' || state == 'd' {
+			log.Errorf("checkDomainHealthy: Domain %d not healthy. Domain state: %s", domainID, domainState)
+			return fmt.Errorf("domain is shutting down")
```

Comment:
This should not be an error.
When this happens (or much better, if neither b nor r is set), then you shouldn't bother looking for a qemu process since the domain is in an indeterminate or transient state.

You can tell that it is good or bad in that case. Hence the confusion.

Reply:
But qemu kill is initiated only when the state reaches either ['s', 'c', 'd'] state and there is no transition after that. That's the reason why I'm looking for those bits to determine whether to check qemu or not. I guess It's safe to assume that qemu will be present until neither of ['s', 'c', 'd'] state is detected.

Reply:
When s, c, or d are set you shouldn't look for a qemu process.
But I'd argue that you shouldn't look for a qemu process when the reported state is ----- because that makes a lot of assumption about what the hypervisor is doing; it could change at any time.

Reply:
> But I'd argue that you shouldn't look for a qemu process when the reported state is ----- because that makes a lot of assumption about what the hypervisor is doing; it could change at any time.

100% agreed with the above. @adarsh-zededa can you please make a change?

Reply:
Added a new method `IsDomainHealthy()` which will return true if the domain is in healthy ('r' or 'b') state. 
Currently `IsDomainHealthy()`returns true by default in kvm.go. Should I add a similar check 
in kvm.go as well?